### ISSUE TITLE
installing requirements for start page fails: In --require-hashes mode, all requirements must have their versions pinned with ==

### ISSUE THREAD

**Issue by mkmelin** [thundernest/thunderbird-website#53]
Setting up to build the start page I get

pip install -r requirements.txt

In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
    setuptools from https://files.pythonhosted.org/packages/ff/f4/385715ccc461885f3cedf57a41ae3c12b5fec3f35cce4c8706b1a112a133/setuptools-40.0.0-py2.py3-none-any.whl#sha256=d68abee4eed409fbe8c302ac4d8429a1ffef912cd047a903b5701c024048dd49 (from icalendar==3.8.3->-r requirements.txt (line 28))

**Comment by Sancus** [comment]
What platform is this on? I'm legit confused how you can even have pip installed without setuptools.

**Comment by mkmelin** [comment]
Ubuntu 18.04.

**Comment by mkmelin** [comment]
Are you saying I should just install python-setuptools?

**Comment by mkmelin** [comment]
Eh, it's already installed
"python-setuptools is already the newest version (39.0.1-2)."

**Comment by Sancus** [comment]
I think you need to upgrade it to 40.0, not really sure why since icalendar 3.8.3 installs fine for me on 39 but you should be able to `pip install --upgrade setuptools`.

**Comment by mkmelin** [comment]
Uninstalled python-setuptools, installed setuptools-40.0.0 through pip. Still the same issue.

**Comment by Sancus** [comment]
OK, well, you managed to trigger a really strange bug related to the fact that [pip does not have a real dependency resolver and is thus non-deterministic](https://github.com/mozilla/bedrock/issues/5967#issuecomment-410051028).

This means that sometimes it will try to install sub-dependencies, and sometimes it won't, depending on the order of package installation.

I've updated to icalendar==4.0.2 which removes the redundant(and strangely harmful) setuptools requirement, and tested on Ubuntu18 and requirements install correctly now.

**Comment by mkmelin** [comment]
Thanks for sorting it out!

**Comment by ryanleesipes** [comment]
This is happening again. Any way to ensure this doesn't break?

**Comment by Sancus** [comment]
> This is happening again. Any way to ensure this doesn't break?

No, not really. pip still doesn't have a real dependency resolver. Work is happening gradually though.

This was mostly because I added flake8 and the tool I was using to automatically add hashes doesn't seem like it did a good job. pip-tools may work better, will try using that going forward. It fixed the issue, anyway.

Fixed in https://github.com/thundernest/thunderbird-website/commit/87778a071b460eeb9f31d6af39cd888958ebd415
